tests/test_migrators.py

```diff
@@ -551,6 +553,94 @@ def run_test_migration(
     return pmy
 
 
+def run_test_migration_recipe_yaml(
+    migrator: Migrator,
+    in_yaml: str,
+    output: str,
+    kwargs: dict,
+    prb: str,
+    mr_out: dict,
+    tmp_path: Path,
+    should_filter: bool = False,
+    make_body: bool = False,
+):
+    if mr_out:
+        mr_out.update(bot_rerun=False)
+
+    tmp_path.joinpath("recipe.yaml").write_text(in_yaml)
+
+    # read the conda-forge.yml
+    cf_yml_path = tmp_path.parent / "conda-forge.yml"
+    cf_yml = cf_yml_path.read_text() if cf_yml_path.exists() else "{}"
+
+    name = yaml.safe_load(in_yaml)["package"]["name"]
+    pmy = populate_feedstock_attributes(
+        name, sub_graph={}, recipe_yaml=in_yaml, conda_forge_yaml=cf_yml
+    )
+
+    # these are here for legacy migrators
+    pmy["version"] = pmy["meta_yaml"]["package"]["version"]
```

Comment:
This function started out as a copy of `run_test_migration` the `meta.yaml` pendant. Only up to this point, I am fairly sure that nothing needs to be adapted anymore. The rest should be evaluated if it is needed or if it needs to be adapted.